### ISSUE TITLE
Add natural language search

### ISSUE THREAD

**Issue by jasonjmcghee** [jasonjmcghee/xrem#11]
Leverage something like `sqlite-vss` to add natural language search (in addition to) FTS.

The embedding function via `candle` already exists in the code.

**Comment by Thawab8** [comment]
duckdb has a vector search now: 
https://duckdb.org/2024/05/03/vector-similarity-search-vss.html

**Comment by jasonjmcghee** [comment]
Indeed! Would be very cool to give it a go. Any interest in giving it a shot?

**Comment by Thawab8** [comment]
hahaha, I wish that i can code in rust. This is a project that uses duckdb as a vector store. xrem might use GraphRag instead of the regular chuck indexing. 
https://github.com/bradAGI/GraphMemory
brad is active on twitter, i do believe he will build something great.
